### ISSUE TITLE
xgboost multiclass classification problem

### ISSUE THREAD

**Issue by weidong8405** [dmlc/xgboost#4840]
Now the xgboost training if the objective function is multi:softmax, it will training each class one tree, so in this case if i want use 50 trees to boost or parallel, in deed i use 50 * num_class trees, that may cost a lot in inference.
So my question is whether xgboost support real multi classification, use one tree to predict all the class's probility?

**Comment by trivialfis** [comment]
No.  It doesn't.  Recommendation for related papers are welcomed.

**Comment by vengodelsur** [comment]
Related papers:
SAMME (multiclass AdaBoost) https://web.stanford.edu/~hastie/Papers/samme.pdf 
Tensorflow Boosted trees implementation https://arxiv.org/abs/1710.11547

**Comment by trivialfis** [comment]
https://github.com/dmlc/xgboost/issues/4972

**Comment by MrRace** [comment]
for me, class num is 33, do the mutil-label text classification, in the inference stage, it will cost a minute for a input text. What should I do to accelerate the inference? Thanks a lot!
